tsdb/engine/tsm1/engine_test.go

```diff
@@ -250,6 +251,274 @@ func TestEngine_Backup(t *testing.T) {
+	}
+	ex1Data, err := getExportData(&ex1)
+	if err != nil {
+		t.Errorf("Error extracting data from export: %s", err.Error())
+	}
+
+	for k, v := range ex1Data {
+		fullExp, ok := fileData[k]
+		if !ok {
+			t.Errorf("Extracting subset resulted in file not found in full export: %s", err.Error())
+			continue
+		}
+		if !equalBuffers(fullExp, v) {
+			t.Errorf("2Difference in data between backup and Export for file %s", th.Name)
+		}
+
+	}
+
+	var ex2 bytes.Buffer
+	if err := e.Export(&ex2, "", time.Unix(0, 1000000001), time.Unix(0, 2000000000)); err != nil {
+		t.Fatalf("failed to export: %s", err.Error())
+	}
+
+	ex2Data, err := getExportData(&ex2)
+	if err != nil {
+		t.Errorf("Error extracting data from export: %s", err.Error())
+	}
+
+	for k, v := range ex2Data {
+		fullExp, ok := fileData[k]
+		if !ok {
+			t.Errorf("Extracting subset resulted in file not found in full export: %s", err.Error())
+			continue
+		}
+		if !equalBuffers(fullExp, v) {
+			t.Errorf("2Difference in data between backup and Export for file %s", th.Name)
+		}
+
+	}
+
+	var ex3 bytes.Buffer
+	if err := e.Export(&ex3, "", time.Unix(0, 2000000001), time.Unix(0, 3000000000)); err != nil {
+		t.Fatalf("failed to export: %s", err.Error())
+	}
+
+	ex3Data, err := getExportData(&ex3)
+	if err != nil {
+		t.Errorf("Error extracting data from export: %s", err.Error())
+	}
+
+	for k, v := range ex3Data {
+		fullExp, ok := fileData[k]
+		if !ok {
+			t.Errorf("Extracting subset resulted in file not found in full export: %s", err.Error())
+			continue
+		}
+		if !equalBuffers(fullExp, v) {
+			t.Errorf("2Difference in data between backup and Export for file %s", th.Name)
+		}
+
+	}
+
+	var ex12 bytes.Buffer
+	if err := e.Export(&ex12, "", time.Unix(0, 0), time.Unix(0, 2000000000)); err != nil {
+		t.Fatalf("failed to export: %s", err.Error())
+	}
+
+	ex12Data, err := getExportData(&ex12)
+	if err != nil {
+		t.Errorf("Error extracting data from export: %s", err.Error())
+	}
+
+	for k, v := range ex12Data {
+		fullExp, ok := fileData[k]
+		if !ok {
+			t.Errorf("Extracting subset resulted in file not found in full export: %s", err.Error())
+			continue
+		}
+		if !equalBuffers(fullExp, v) {
+			t.Errorf("2Difference in data between backup and Export for file %s", th.Name)
+		}
+
+	}
+
+	var ex23 bytes.Buffer
+	if err := e.Export(&ex23, "", time.Unix(0, 1000000001), time.Unix(0, 3000000000)); err != nil {
+		t.Fatalf("failed to export: %s", err.Error())
+	}
+
+	ex23Data, err := getExportData(&ex23)
+	if err != nil {
+		t.Errorf("Error extracting data from export: %s", err.Error())
+	}
+
+	for k, v := range ex23Data {
+		fullExp, ok := fileData[k]
+		if !ok {
+			t.Errorf("Extracting subset resulted in file not found in full export: %s", err.Error())
+			continue
+		}
+		if !equalBuffers(fullExp, v) {
+			t.Errorf("2Difference in data between backup and Export for file %s", th.Name)
+		}
+
+	}
+}
+
+func equalBuffers(bufA, bufB *bytes.Buffer) bool {
```

Comment:
Is there a reason you didn't use https://golang.org/pkg/bytes/#Equal here on the `Bytes()` calls? If `bufB.Len() < bufA.Len()`, this will panic. Also, it is only checking that `bufB` has a prefix of `bufA`, not that they are the same length and have the same contents.